File: Java/src/main/java/com/example/Tank.java
package com.example;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class Tank {
    private String direction;
    private float pwm;
    private float leftValue;
    private float rightValue;

    public Tank() {
        this("stop", 100, 0, 0);
    }

    public void update(String direction, float pwm) {
        this.direction = direction;
        this.pwm = pwm;
    }

    public void evaluateVelocity() {

        switch(direction) {
            case("forward"):
                leftValue = pwm;
                rightValue = pwm;
                break;

            case("left"):
                leftValue = -pwm;
                rightValue = pwm;
                break;

            case("right"):
                leftValue = pwm;
                rightValue = -pwm;
                break;

            case("backwards"):
                leftValue = -pwm;
                rightValue = -pwm;
                break;

            default:
                leftValue = 0;
                rightValue = 0;
        }
    }

    public TankResponse getResponse() {
        return new TankResponse(leftValue, rightValue);
    }
}

File: Java/src/main/resources/static/js/tank.js
const url = 'http://192.168.0.167:80/update';
let pwmValue = 100;

const data = {
    forward: {
        direction: 'forward',
        pwm: pwmValue
    },

    left: {
        direction: 'left',
        pwm: pwmValue
    },

    right: {
        direction: 'right',
        pwm: pwmValue
    },

    backwards: {
        direction: 'backwards',
        pwm: pwmValue
    },

    stop: {
        direction: 'stop',
        pwm: pwmValue
    }
}

function getOptions(value) {
    return {
        method: 'POST',
        body: JSON.stringify(data[value]),
        headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json'
        }
    }
}

document.getElementById('forward').addEventListener('click', (e) => {
    fetch(url, getOptions('forward'))
    .then(response => response.json())
    .then(response => console.log(response))
});

document.getElementById('left').addEventListener('click', (e) => {
    fetch(url, getOptions('left'))
    .then(response => response.json())
    .then(response => console.log(response))
});

document.getElementById('right').addEventListener('click', (e) => {
    fetch(url, getOptions('right'))
    .then(response => response.json())
    .then(response => console.log(response))
});

document.getElementById('backwards').addEventListener('click', (e) => {
    fetch(url, getOptions('backwards'))
    .then(response => response.json())
    .then(response => console.log(response))
});

document.getElementById('stop').addEventListener('click', (e) => {
    fetch(url, getOptions('stop'))
    .then(response => response.json())
    .then(response => console.log(response))
});